conditional-branch.bnf

```diff
@@ -0,0 +1,68 @@
+# Case
+
+<case_expr> ::= case <patterns> do <case_clause_exprs> end
+<patterns> ::= <list> | <tuple> | <term>
+<case_clause_exprs> ::= <case_clause_expr> | <case_clause_expr><case_clause_exprs>
+<case_clause_expr> ::= <case_condition_expr> -> <case_result>
+<case_condition_expr> ::= _ | <integer> | <float> | <alpha> | <guards> | :error | <term> | ^<term>
+<guards> ::= <guard_clause> when <guards_expr>
+<guards_expr> ::= <comparison_expr> 
+    | <type_checks> 
+    | <boolean_strict_expr> 
+    | <arithmetic_expr> 
+    | <unary_expr>
+    | <in_comparison_operator>
+    | <not_in_comparison_operator>
+    | <bitwise_expr>
+    | <term>
```

Comment:
yang ini minta link referensinya dong rits

Reply:
Ini referensi buat [Guards](https://hexdocs.pm/elixir/patterns-and-guards.html#guards)

Reply:
Not Harits, tapi referensi soal <term> ada di https://hexdocs.pm/elixir/typespecs.html#built-in-types.
Jadi term() equivalen dengan any(), lalu any() itu bisa diisi sembarang basic type.

Reply:
`<term>` berarti ga perlu ya dalam guards karena bukan termasuk dalam klausanya

Reply:
@IvanAldion kalo `<term>` ditaruh sebagai element pada tuple dan list valid ga secara syntax?

Reply:
ini konteks kita <term> teh apa ya? 
kalau yang urang tangkep obrolan ini teh <term> berarti nilai yang valid di elixir ya, misal
```
1
:abc
Abc.Abc
[1,2,3]
dst..
```
skalian ngangkat, buat kata <identifier> buat identifier yang dibikin sama user bahasanya, kayak misal nama variable, nama fungsi, etc.

Reply:
nah iya itu, nama rulenya apa ya? identifier?

Reply:
@otakbeku Sepertinya valid, soalnya `<term>` kalau diturunin ujungnya nanti jadi 1 tipe (bisa atom/int/float/list etc). 
Ini bisa di compile `[[1,2.0,'3',true],4,5.0,'6',false]` .

Reply:
oh alright gab! aman berarti